client/state/sites/selectors/get-site.js

```diff
@@ -12,10 +12,13 @@ let getSiteCache = new WeakMap();
  * Returns a normalized site object by its ID or site slug.
  *
  * @param  {object}  state  Global state tree
- * @param  {number|string}  siteIdOrSlug Site ID or site slug
- * @returns {import('calypso/state/ui/selectors/site-data').SiteData|null|undefined}        Site object
+ * @param  {number|string|undefined}  siteIdOrSlug Site ID or site slug
+ * @returns {import('@automattic/data-stores').SiteDetails|null|undefined}        Site object
  */
 export default function getSite( state, siteIdOrSlug ) {
+	if ( ! siteIdOrSlug ) {
+		return null;
```

Comment:
`getSite()` already returned null if there was no site, this just makes it explicit.

Reply:
The only downside of this would be if we wanted to have some sort of a "fallback site" returned by the following `rawSite` retrieval code, however I don't think we're heading that way, so we should be good.